app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,95 @@
+
+class TasksController < ApplicationController
+  
+  def index
+    @tasks = Task.all
+  end
+  
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+    
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+  
+  def new
+    @task = Task.new
+  end
+  
+  def create
+    @task = Task.new(
+      name: params[:task][:name], 
+      description: params[:task][:description], 
+      completion_date: params[:task][:completion_date]) #instantiate a new task
+      if @task.save # save returns true if the database insert succeeds
+        redirect_to task_path(@task.id) # go to the index so we can see the task in the list
+        return
+      else # save failed :(
+        render :new # show the new task form view again
+        return
+      end
+    end
+    
+    def edit
+      @task = Task.find_by(id: params[:id])
+      
+      if @task.nil?
+        redirect_to tasks_path
+        return
+      end
+    end
+    
+    def update
+      @task = Task.find_by(id: params[:id])
+      if @task.nil? 
+        redirect_to tasks_path
+      else 
+        @task.update(
+          name: params[:task][:name], 
+          description: params[:task][:description],
+          completion_date: params[:task][:completion_date],
+          completed: params[:task][:completed]
+        )
+        redirect_to tasks_path # go to the index so we can see the task in the list
+      end
+      #   return
+      # else # save failed :(
+      #   render :edit # show the new task form view again
+      #   return
+    end
+    
+    def complete
+      @task = Task.find_by(id: params[:id])
+      if @task[:completed] == false
```

Comment:
Your tasks start off with this field nil because you don't explicitly include `completed: false` in the `create` controller action, so you have to mark complete twice for the task to be listed as completed.